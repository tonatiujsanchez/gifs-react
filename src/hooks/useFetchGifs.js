import { useState, useEffect } from "react"
import { getGif } from "../helpers/getGifs"



const useFetchGifs = (category ) => {


    const [ state, setState ] = useState({
        data: [],
        loading: true
    })

    useEffect(()=>{

        getGif(category).then(
            gifs => {

                setTimeout(()=>{
                    setState( {
                        data: [ ...gifs ],
                        loading: false
                    })
                },1500)

            } 
        )

    },[ category ])



  return state // { data:[], loading: true }
}

export default useFetchGifs