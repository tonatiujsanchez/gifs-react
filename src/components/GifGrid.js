import GifGridItem from "./GifGridItem"
import useFetchGifs from "../hooks/useFetchGifs"
import Loading from "./Loading"

import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category )

  return (
      <div className="category">
        <h3>{ category }</h3>
        { loading 
            ? <Loading />
            : <div className="card-grid">
                { data.map( image =>  <GifGridItem key={ image.id } { ...image } /> ) }
             </div>
        }
      </div>
    )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid