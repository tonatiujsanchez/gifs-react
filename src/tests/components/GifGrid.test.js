import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"


import useFetchGifs from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')




describe('Pruebas del componente GifGrid', () => { 

    const category = 'Trunks'
    
    
    test('El componente GifGrid debe de mostrarse correctamente', () => { 
        
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        let wrapper = shallow( <GifGrid category={ category } /> )
        
        expect( wrapper ).toMatchSnapshot()

     })



     test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => { 
        
        const gifs = [{
            id: 'ABC',
            img: 'https://giphy.com/',
            title: 'Title of gif'
        },
  ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        let wrapper = shallow( <GifGrid category={ category } /> ) 

        
        const loading = wrapper.find('Loading').exists()
        expect( loading ).toBe( false )


        const GifGridItem = wrapper.find('GifGridItem')
        expect( GifGridItem.length ).toBe( gifs.length ) 
     })


 })



