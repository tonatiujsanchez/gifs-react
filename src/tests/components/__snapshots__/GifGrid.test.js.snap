// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid El componente GifGrid debe de mostrarse correctamente 1`] = `
<div
  className="category"
>
  <h3>
    Trunks
  </h3>
  <Loading />
</div>
`;
